### DISCUSSION TITLE
Understanding multi-user abilities

### DISCUSSION THREAD

**Discussion by chrisroat** [Project-MONAI/MONAILabel#1543] [Q&A] [answered]
Hi,

I am evaluating various dicom labeling solutions, and MONAI looks interesting because of its active development and because it is used/mentioned with many cloud providers.

In the multi-user setup, is it possible to have per-user labels, and then to define a specific label to be used for an image during training?

**Comment by tangy5** [comment]
Can you explain more about "per-user labels". MONAI and MONAI Label apps and models can have customized/specific labels and the created/annotated datasets can be used for training. Multiple-users  can work on same image and continue to annotate until the final segmentation/label is done and saved to dicom-web.

**Reply by chrisroat** [reply]
Sure thing.  I am asking which of these two situations exist:
- One label per image.  Multiple users annotating/updating the same label for an image.
- One label per image per user.  Each user creates there own label for an image.

If the latter (which I am hoping), then:
- Can a user see another users labels?
- Can an admin see all the users labels?
- When making a dataset, can single user's label be specified for the image?

**Comment by tangy5** [comment, marked as answer]
The requirement is that you will need multiple-users to work on same image. 
Ok, I feel you will need some label version management logic to meet your requirements:

**1. One label per image. Multiple users annotating/updating the same label for an image.**
Yes. Once one user finished annotation, save it as the "original" label, and then the second user can see and load the "original" label to continue to annotate. Once all users finish annotation, the label can be saved to "final" for training.  This can be achieved, just be aware of the label version control, if two users save label at the same time, one of the labels may be overridden if there is no version control logic.
**2. One label per image per user. Each user creates there own label for an image.**
This is currently not the natural case for MONAI Label, 
Becuase there will be only one label associated with the image for training eventually, the default MONAI Label will save the latest annotation to a folder named "final". If there are multiple users who work on the same image and resulting in multiple labels, you will need to decide which label to put in the "final" folder.  Or you can make some code changes to save different versions of labels. 
A bypass option, you can copy/duplicate the datasets and set two MONAI Label servers for two users, the labels won't conflict and save to two filesystem folders.

In summary, if you want to do dataset version control for same image labels, you will need some customization and code changes. otherwise, it's totally ok if multiple users work on different images in the dataset.

**Reply by chrisroat** [reply]
Thank you for taking the time to address my concerns.  Our use case is that each user would create a private label for each image, not seeing what other users do.  I appreciate your input.

**Reply by tangy5** [reply]
No problem. MONAI Label can create private label for an image, each user won't see other's label. But as mentioned above, if you would like to save and version multiple labels to the same image. There will be some minor code changes for datastore logic.

**Reply by SachidanandAlle** [reply]
You can create multiple labels for the same image.. and save under different tags..

Only `final` tagged labels will be picked for training or fine-tuning the model


Otherwise you can create as many as.. and each label can also hold extra meta information where you can save any information about annotator etc.. please check the API for saving the label.. you might want to play around with that API to fit your usecase
